src/identity-providers/add/SamlConnectSettings.tsx

```diff
@@ -0,0 +1,197 @@
+              providerId: id,
+              fromUrl: entityUrl,
+            });
+          } catch (error) {
+            result = { error };
+          }
+          setDiscoveryResult(result as Result);
+          setupForm(result);
+          setDiscovering(false);
+        })();
+    }
+  }, [discovering]);
+
+  const fileUpload = async (obj: object) => {
+    if (obj) {
+      const formData = new FormData();
+      formData.append("providerId", id);
+      formData.append("file", new Blob([JSON.stringify(obj)]));
+
+      try {
+        const response = await fetch(
+          `${getBaseUrl(
+            adminClient
+          )}admin/realms/${realm}/identity-provider/import-config`,
+          {
+            method: "POST",
+            body: formData,
+            headers: {
+              Authorization: `bearer ${await adminClient.getAccessToken()}`,
+            },
+          }
+        );
+        const result = await response.json();
+        setupForm(result);
+      } catch (error) {
+        setDiscoveryResult({ error });
+      }
+    }
+  };
+
+  return (
+    <>
+      <Title headingLevel="h4" size="xl" className="kc-form-panel__title">
+        {t("samlSettings")}
+      </Title>
+
+      <FormGroup
+        label={t("serviceProviderEntityId")}
+        fieldId="kc-service-provider-entity-id"
+        labelIcon={
+          <HelpItem
+            helpText="identity-providers-help:serviceProviderEntityId"
+            forLabel={t("serviceProviderEntityId")}
+            forID="kc-service-provider-entity-id"
+          />
+        }
+        isRequired
+      >
+        <TextInput
+          type="text"
+          name="config.entityId"
+          data-testid="serviceProviderEntityId"
+          id="kc-service-provider-entity-id"
+          value={entityUrl || defaultEntityUrl}
+          onChange={setEntityUrl}
+          ref={register({ required: true })}
+        />
+      </FormGroup>
+
+      <FormGroup
+        label={t("useEntityDescriptor")}
+        fieldId="kc-use-entity-descriptor"
+        labelIcon={
+          <HelpItem
+            helpText="identity-providers-help:useEntityDescriptor"
+            forLabel={t("useEntityDescriptor")}
+            forID="kc-use-entity-descriptor-switch"
+          />
+        }
+      >
+        <Switch
+          id="kc-use-entity-descriptor-switch"
+          label={t("common:on")}
+          data-testid="useEntityDescriptor"
+          labelOff={t("common:off")}
+          isChecked={descriptor}
+          onChange={setDescriptor}
+        />
+      </FormGroup>
+
+      {descriptor && (
+        <FormGroup
+          label={t("samlEntityDescriptor")}
+          fieldId="kc-saml-entity-descriptor"
+          labelIcon={
+            <HelpItem
+              helpText="identity-providers-help:samlEntityDescriptor"
+              forLabel={t("samlEntityDescriptor")}
+              forID="kc-saml-entity-descriptor"
+            />
+          }
+          isRequired
```

Comment:
This will make the field red with a hint when the user doesn't fill it
```suggestion
          isRequired
          helperTextInvalid={t("common:required")}
          validated={
            errors.samlEntityDescriptor ? ValidatedOptions.error : ValidatedOptions.default
          }
```